### PR TITLE
Readable underwiting config

### LINKED ISSUE #7: underwriting should have human-readable config
current yml config for underwriting is awful. write a new one.
it should store/allow
- [x] start and end dates
- [x] arbitrary metadata about underwriting (e.g. transcripts)
- [x] multiple recordings for each underwriting
- [ ] multiple underwriters per hour

Ideally, underwriting will play every underwriter for a given hour, and will select a random underwriting recording if there are multiple for a single piece of underwriting. It also should verify that it is before the end date and after the start date, so that we don't need to babysit underwriting. Some less useful potential features: use liquidsoap's built-in `say` support to read transcripts, if we have no recording of the underwriting.

Long-term, it may be possible to adapt this, with minimal modifications, to use for scheduled PSAs, Promos, and Rotation.

### PR DESCRIPTION
**Pull request by echarlie**
Should mostly fix #7 

Probably some erroneous or superfluous changes in there, but my limited testing suggests it does work. I also provide a sample underwriting config file, so that one doesn't have to look at the code to figure out how it works.

### CODE REVIEW

**Comment by mutantmonkey** [comment]
Please follow PEP8 formatting. `flake8` reports:
```
queue_underwriting.py:15:1: E302 expected 2 blank lines, found 1
queue_underwriting.py:32:1: E302 expected 2 blank lines, found 1
queue_underwriting.py:35:9: E265 block comment should start with '# '
queue_underwriting.py:39:1: E305 expected 2 blank lines after class or function definition, found 1
queue_underwriting.py:42:10: E225 missing whitespace around operator
queue_underwriting.py:43:12: E225 missing whitespace around operator
queue_underwriting.py:45:9: E201 whitespace after '('
queue_underwriting.py:45:11: E201 whitespace after '('
queue_underwriting.py:45:79: E202 whitespace before ')'
queue_underwriting.py:45:80: E501 line too long (85 > 79 characters)
queue_underwriting.py:45:85: E502 the backslash is redundant between brackets
queue_underwriting.py:46:14: E201 whitespace after '('
queue_underwriting.py:46:78: E202 whitespace before ')'
queue_underwriting.py:46:80: E501 line too long (86 > 79 characters)
queue_underwriting.py:46:80: E202 whitespace before ')'
queue_underwriting.py:47:80: E501 line too long (105 > 79 characters)
queue_underwriting.py:50:60: E231 missing whitespace after ','
queue_underwriting.py:50:80: E501 line too long (88 > 79 characters)
queue_underwriting.py:52:80: E501 line too long (82 > 79 characters)
queue_underwriting.py:54:80: E501 line too long (95 > 79 characters)
queue_underwriting.py:55:9: E265 block comment should start with '# '
queue_underwriting.py:61:46: E231 missing whitespace after ','
queue_underwriting.py:65:78: E202 whitespace before ')'
queue_underwriting.py:65:80: E501 line too long (81 > 79 characters)
queue_underwriting.py:65:80: E202 whitespace before ')'
```

File: queue_underwriting.py
Comment (mutantmonkey) [inline queue_underwriting.py]: It would be better to use `random.choice` here.

File: queue_underwriting.py
Comment (mutantmonkey) [inline queue_underwriting.py]: Why did you remove the time from this message? It'd be helpful to have it logged.

File: sample_underwriting.yml
Comment (mutantmonkey) [inline sample_underwriting.yml:20]: `/dev/urandom` is not a good URL to use.

File: queue_underwriting.py
Comment (mutantmonkey) [inline queue_underwriting.py]: This is not valid syntax. You probably want `sys.exit` instead.

File: queue_underwriting.py
Comment (mutantmonkey) [inline queue_underwriting.py]: Please log the hour in this message.

File: queue_underwriting.py
Comment (mutantmonkey) [inline queue_underwriting.py]: Please log the hour in this message.

**Review by mutantmonkey — CHANGES_REQUESTED** [review]

File: queue_underwriting.py
Comment (echarlie) [reply queue_underwriting.py]: looking at the docs, that seems more idiomatic; will do.

File: queue_underwriting.py
Comment (echarlie) [reply queue_underwriting.py]: oversight in trying to get stuff to work; I'll re-add it.

File: sample_underwriting.yml
Comment (echarlie) [reply sample_underwriting.yml:20]: it's an arbitrary example to test parsing. I don't see why there's an issue

**Comment by mutantmonkey** [comment]
There are still formatting issues:
```
playlist-loader/playlist.py:53:1: E302 expected 2 blank lines, found 1
playlist-loader/playlist.py:59:21: E711 comparison to None should be 'if cond is not None:'
playlist-loader/playlist.py:72:13: E722 do not use bare except'
playlist-loader/loader.py:53:9: E722 do not use bare except'
playlist-loader/loader.py:77:1: E305 expected 2 blank lines after class or function definition, found 1
```

**Comment by echarlie** [comment]
the playlist-loader file wasn't edited in this pr; the problems exist upstream.

**Review by mutantmonkey — APPROVED** [review]

**Comment by mutantmonkey** [comment]
Ah, sorry about that. That should be fixed at some point though.

I'd like to get another reviewer before we merge.

**Comment by echarlie** [comment]
I'll write up another PR that makes everything flake8-approved.

**Pull request merged**
